dev-proxy-abstractions/ILogger.cs

```diff
@@ -40,15 +41,20 @@ public enum LogLevel
     Error
 }
 
-public interface ILogger : ICloneable
+/// <summary>
+/// Custom interface for logging, extending <see cref="Microsoft.Extensions.Logging.ILogger"/>
+/// </summary>
+/// <remarks>
+/// Please use structured logging as much as possible.
+/// Log Debug information with <see cref="Microsoft.Extensions.Logging.ILogger.LogDebug(string, object[])"/>
+/// Log Information with <see cref="Microsoft.Extensions.Logging.ILogger.LogInformation(string, object[])"/>
+/// Log Warnings with <see cref="Microsoft.Extensions.Logging.ILogger.LogWarning(string, object[])"/>
+/// Log Errors with <see cref="Microsoft.Extensions.Logging.ILogger.LogError(Exception, string, object[])"/> or <see cref="Microsoft.Extensions.Logging.ILogger.LogError(string, object[])"/>
+/// </remarks>
+public interface ILogger : ICloneable, MSLogging.ILogger
 {
-    public LogLevel LogLevel { get; set; }
+    public void SetLogLevel(LogLevel logLevel);
```

Comment:
Because with structured logging the whether or not a specific level is enabled usually comes from a runtime dynamic configuration. But for this proxy which only partially uses structured logging it makes sense to change the level on the logger itself.

Reply:
This still doesn't explain why we need a method to change it and can't use the existing property. What am I missing?

Reply:
What do you think of removing the LogLevel from the dev proxy and just use the LogLevel from  Microsoft.Extensions.Logging? How does that work when loading it from settings?

Reply:
This [code](https://github.com/microsoft/dev-proxy/blob/f64d765a3f0099941942ae9412c74ba5ebe6848a/dev-proxy/ProxyConfiguration.cs#L32-L33) parses `debug`, `info`, `warn` and `error`. That will no longer work, does that then require a modification in the docs? Or do we keep the old enum to set the value and convert it in `SetLogLevel(..)` where it's converted to `Microsoft.Extensions.Logging.LogLevel`?

Reply:
I'm okay with aligning with `Microsoft.Extensions.Logging.LogLevel`. When we merge this PR, we'll update docs accordingly.

Reply:
removed